src/nspawn/nspawn.c

```diff
@@ -3206,6 +3206,17 @@ static int outer_child(
         return 0;
 }
 
+static int on_sigchld(sd_event_source *s, const struct signalfd_siginfo *si, void *userdata) {
+        pid_t pid;
+
+        pid = PTR_TO_UINT32(userdata);
+        if (pid == si->ssi_pid) {
+                sd_event_exit(sd_event_source_get_event(s), 0);
+        }
+
+        return 0;
```

Comment:
We should really reap those other processes here, otherwise they become zombies and are never cleaned up. It would also be good to add a comment here why they might exist. i.e. say that PR_SET_CHILD_REAPER causes this.